java/ql/lib/semmle/code/java/frameworks/apache/IOGenerated.qll

```diff
@@ -1,684 +1,828 @@
-        "org.apache.commons.io;FileUtils;true;writeLines;(File,Collection,String);;Argument[0];create-file",
-        "org.apache.commons.io;FileUtils;true;writeLines;(File,Collection,String,boolean);;Argument[0];create-file",
-        "org.apache.commons.io;FileUtils;true;writeLines;(File,Collection,boolean);;Argument[0];create-file",
-        "org.apache.commons.io;FileUtils;true;writeLines;(File,String,Collection);;Argument[0];create-file",
-        "org.apache.commons.io;FileUtils;true;writeLines;(File,String,Collection,String);;Argument[0];create-file",
-        "org.apache.commons.io;FileUtils;true;writeLines;(File,String,Collection,String,boolean);;Argument[0];create-file",
-        "org.apache.commons.io;FileUtils;true;writeLines;(File,String,Collection,boolean);;Argument[0];create-file",
-        "org.apache.commons.io;FileUtils;true;writeStringToFile;(File,String);;Argument[0];create-file",
-        "org.apache.commons.io;FileUtils;true;writeStringToFile;(File,String,Charset);;Argument[0];create-file",
-        "org.apache.commons.io;FileUtils;true;writeStringToFile;(File,String,Charset,boolean);;Argument[0];create-file",
-        "org.apache.commons.io;FileUtils;true;writeStringToFile;(File,String,String);;Argument[0];create-file",
-        "org.apache.commons.io;FileUtils;true;writeStringToFile;(File,String,String,boolean);;Argument[0];create-file",
-        "org.apache.commons.io;FileUtils;true;writeStringToFile;(File,String,boolean);;Argument[0];create-file",
-        "org.apache.commons.io;IOUtils;true;copy;(URL,File);;Argument[0];open-url",
-        "org.apache.commons.io;IOUtils;true;copy;(URL,File);;Argument[1];create-file",
-        "org.apache.commons.io;IOUtils;true;copy;(URL,OutputStream);;Argument[0];open-url",
-        "org.apache.commons.io;IOUtils;true;toByteArray;(URI);;Argument[0];open-url",
-        "org.apache.commons.io;IOUtils;true;toByteArray;(URL);;Argument[0];open-url",
-        "org.apache.commons.io;IOUtils;true;toString;(URI);;Argument[0];open-url",
-        "org.apache.commons.io;IOUtils;true;toString;(URI,Charset);;Argument[0];open-url",
-        "org.apache.commons.io;IOUtils;true;toString;(URI,String);;Argument[0];open-url",
-        "org.apache.commons.io;IOUtils;true;toString;(URL);;Argument[0];open-url",
-        "org.apache.commons.io;IOUtils;true;toString;(URL,Charset);;Argument[0];open-url",
-        "org.apache.commons.io;IOUtils;true;toString;(URL,String);;Argument[0];open-url",
-        "org.apache.commons.io;RandomAccessFileMode;false;create;(File);;Argument[0];create-file",
-        "org.apache.commons.io;RandomAccessFileMode;false;create;(Path);;Argument[0];create-file",
-        "org.apache.commons.io;RandomAccessFileMode;false;create;(String);;Argument[0];create-file"
+        "org.apache.commons.io.file;PathFilter;true;accept;(Path,BasicFileAttributes);;Argument[-1];generated:regex-use[0]",
```

Comment:
The `regex-use` sinks that include a number are meant to convey some additional information about the predicate they're used on, in particular that the specified argument is the string that the regex is matched against. As such they should probably not be included in generated models. 

Additionally, the queries that use them only consider value flow and not taint flow; so generated models of `regex-use` sinks that included taint flow may not be correct.

Reply:
Lets just avoid creating new sinks based on regex-use then. Thank you @joefarebrother !